selenium/common/webdriver.rb

```diff
@@ -440,22 +440,23 @@ def select_option(value)
 
 
       def select_option_with_text(value)
-        self.click
-        self.find_elements(:tag_name => "option").each do |option|
-          if option.text === value
-            Selenium::Config.retries.times do |try|
-              return if option.attribute('selected')
-
-              option.click
-              sleep 0.1
-            end
+        passed = true 
```

Comment:
Shouldn't this be passed = false? I don't see where passed is ever set to false.

Reply:
Yikes. 
Yes, this is really sloppy, especially since it's replicating a lot of the stuff find_element is doing. I've fixed to to just use find_element to do the testing if it exists or not.